types/react-virtualized/dist/es/WindowScroller.d.ts

```diff
@@ -6,31 +6,25 @@ import { Validator, Requireable, PureComponent } from "react";
  */
 export const IS_SCROLLING_TIMEOUT = 150;
 
+export type onChildScroll = {
+  scrollTop: number;
+}
+
 export type WindowScrollerChildProps = {
     height: number;
     width: number;
     isScrolling: boolean;
     scrollTop: number;
-    onChildScroll: () => void;
+    onChildScroll: ({ scrollTop: number }: onChildScroll) => void;
```

Comment:
I originally had that, but the linter was throwing an explicit `any` error for the `number` type. I'll post a screenshot when I return the computer.

Reply:
I saw the same error reported here https://github.com/DefinitelyTyped/DefinitelyTyped/issues/24949

Reply:
Ok, here's the exact message without the `onChildScroll` type:

```
Error: .../react-virtualized/dist/es/WindowScroller.d.ts:18:34
ERROR: 18:34  expect  TypeScript@next compile error:
Binding element 'number' implicitly has an 'any' type.
```

Reply:
so now I am at my pc, too, and can write a bit more :).
What I meant with inlining the type definition of the function params is, that in the current case the type definition is a bit weird and duplicated. 
The type `({ scrollTop: number }: onChildScroll) => void;` describes a function which takes a parameter of the shape `onChildScroll` which stands for `{ scrollTop: number }`. So writing `{ scrollTop: number }: onChildScroll` as the definition of the parameter type is duplicated. And it is weird because it stands for read the property `scrollTop` from the first function parameter, which is of type `onChildScroll`, and use it under the name `number`, so in my opinion TypeScript should throw some errors here.

All in all the type 
```ts
(params: { scrollTop: number }) => void;
```
is short enough to be inline and should not any implicit any errors